File: src/lightnap-ng/src/app/core/backend-api/dtos/users/response/notification-dto.ts
import { NotificationStatus } from "../../../../notifications/entities/notification-status";
import { NotificationType } from "../../../../notifications/entities/notification-type";

/**
 * Represents an application notification.
 *
 * @interface Notification
 * @property {number} id - The unique identifier for the notification.
 * @property {Date} timestamp - The date and time when the notification was created.
 * @property {NotificationStatus} status - The current status of the notification.
 * @property {NotificationType} type - The type/category of the notification.
 * @property {any} data - Additional data associated with the notification.
 */
export interface NotificationDto {
  id: number;
  timestamp: Date;
  status: NotificationStatus;
  type: NotificationType;
  data: any;
}

File: src/LightNap.Core/Notifications/Dto/Response/NotificationDto.cs
using LightNap.Core.Data.Entities;

namespace LightNap.Core.Notifications.Dto.Response
{
    /// <summary>
    /// Represents a notification data transfer object.
    /// </summary>
    public class NotificationDto
    {
        /// <summary>
        /// Gets or sets the notification identifier.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Gets or sets the timestamp of the notification.
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// Gets or sets the status of the notification.
        /// </summary>
        public NotificationStatus Status { get; set; }

        /// <summary>
        /// Gets or sets the type of the notification.
        /// </summary>
        public NotificationType Type { get; set; }

        /// <summary>
        /// Gets or sets the additional data associated with the notification.
        /// </summary>
        public Dictionary<string, object> Data { get; set; } = [];
    }
}